src/parse.rs

```diff
@@ -61,17 +62,31 @@ pub fn get_file_change_description_from_file_paths(
     }
 }
 
+fn _make_line_number_vec(line: &str) -> Vec<usize> {
+    let mut numbers = Vec::<usize>::new();
+    let pattern = r"[-+]";
+    let re = Regex::new(&pattern).unwrap();
+
+    for s in re.split(line) {
+        let number = s.split(',').nth(0).unwrap().split_whitespace().nth(0);
+        match number {
+            Some(number) => numbers.push(number.parse::<usize>().unwrap()),
+            None => continue,
+        }
+    }
+    return numbers;
+}
+
 /// Given input like
 /// "@@ -74,15 +74,14 @@ pub fn delta("
-/// Return " pub fn delta("
-pub fn parse_hunk_metadata(line: &str) -> (&str, &str) {
-    let mut iter = line.split("@@").skip(1);
-    let line_number = iter
-        .next()
-        .and_then(|s| s.split('+').nth(1).and_then(|s| s.split(',').next()))
-        .unwrap_or("");
-    let code_fragment = iter.next().unwrap_or("");
-    (code_fragment, line_number)
+/// Return " pub fn delta(" and a vector of line numbers
+pub fn parse_hunk_metadata(line: &str) -> (&str, Vec<usize>) {
+    let pattern = r"@+ (?P<lns>([-+]\d+(?:,\d+)? ){2,4})@+(?P<cf>.*\s?)";
+    let re = Regex::new(&pattern).unwrap();
+    let caps = re.captures(line).unwrap();
+    let line_numbers = _make_line_number_vec(caps.name("lns").unwrap().as_str());
+    let code_fragment = caps.name("cf").unwrap().as_str();
+    return (code_fragment, line_numbers);
```

Comment:
I just learned that building regular expressions like this inside a function called many times is extremely inefficient. This commit on master speeds delta up by almost 2x 2cb3e403e802250308ef05e37e33105ad68eaef6! I suspect if you do the same thing here it will get rid of the performance difference between this branch and master.

In addition to `make benchmark` I've also added `make flamegraph`, for profiling on MacOS, which is sort of fun. (The Makefile assumes that you've cloned https://github.com/brendangregg/FlameGraph and symlinked two of its perl scripts  into your $PATH with the names `flamegraph` and `stackcollapse-sample`.)